### PR TITLE
[Batch] DataPlane update v15.3.0 - 2022-01-01

### PR DESCRIPTION
**Pull request by paterasMSFT**
# All SDK Contribution checklist:

[Swagger Pull Request](https://github.com/Azure/azure-rest-api-specs/pull/16960)

- Added property uploadHeaders to OutputFileBlobContainerDestination.
  - Allows users to set custom HTTP headers on resource file uploads.
  - Array of type HttpHeader (also being added).
- Added boolean property allowTaskPreemption to JobSpecification, CloudJob, JobAddParameter, JobPatchParameter, JobUpdateParameter
  - Mark Tasks as preemptible for higher priority Tasks (requires Comms-Enabled or Single Tenant Pool).
- Replaced comment (title, description, etc.) references of "low-priority" with "Spot/Low-Priority", to reflect new service behavior.
  - No API change required.
  - Low-Priority Compute Nodes (VMs) will continue to be used for User Subscription pools (and only User Subscription pools), as before.
  - Spot Compute Nodes (VMs) will now be used for Batch Managed (and only Batch Managed pools) pools.

This checklist is used to make sure that common guidelines for a pull request are followed.
- [x] **Please open PR in `Draft` mode if it is:**
	- Work in progress or not intended to be merged.
	- Encountering multiple pipeline failures and working on fixes.
- [x] If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md).**
- [x] **The pull request does not introduce [breaking changes](https://github.com/dotnet/corefx/blob/master/Documentation/coding-guidelines/breaking-change-rules.md).**

### [General Guidelines and Best Practices](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#general-guidelines)
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#testing-guidelines)
- [x] Pull request includes test coverage for the included changes.

### [SDK Generation Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#sdk-generation-guidelines)
- [ ] The generate.cmd file for the SDK has been updated with the version of AutoRest, as well as the commitid of your swagger spec or link to the swagger spec, used to generate the code. (Track 2 only)
- [ ] The `*.csproj` and `AssemblyInfo.cs` files have been updated with the new version of the SDK. Please double check nuget.org current release version.

## Additional management plane SDK specific contribution checklist: 
Note: Only applies to `Microsoft.Azure.Management.[RP]` or `Azure.ResourceManager.[RP]`
 
- [ ] Include updated [management metadata](https://github.com/Azure/azure-sdk-for-net/tree/main/eng/mgmt/mgmtmetadata).
- [ ] Update AzureRP.props to add/remove version info to maintain up to date API versions.

### Management plane SDK Troubleshooting
- If this is very first SDK for a services and you are adding new service folders directly under /SDK, please add `new service` label and/or contact assigned reviewer.
- If the check fails at the `Verify Code Generation` step, please ensure:
	- Do not modify any code in generated folders.
	- Do not selectively include/remove generated files in the PR.
	- Do use `generate.ps1/cmd` to generate this PR instead of calling `autorest` directly.
	Please pay attention to the @microsoft.csharp version output after running `generate.ps1`. If it is lower than current released version (2.3.82), please run it again as it should pull down the latest version.
	
	**Note: We have recently updated the PSH module called by `generate.ps1` to emit additional data. This would help reduce/eliminate the Code Verification check error. Please run following command**:

	    `dotnet msbuild eng/mgmt.proj /t:Util /p:UtilityName=InstallPsModules`

### Old outstanding PR cleanup
 Please note:
	If PRs (including draft) has been out for more than 60 days and there are no responses from our query or followups, they will be closed to maintain a concise list for our reviewers.

### CODE REVIEW

**Review by seth-barshay — APPROVED** [review]

File: sdk/batch/Microsoft.Azure.Batch/CHANGELOG.md [hunk 1]
```diff
@@ -1,16 +1,24 @@
...
+  - Array of type `HttpHeader` (also being added).
+- Added boolean property `AllowTaskPreemption` to `JobSpecification`, `CloudJob`, `JobAddParameter`, `JobPatchParameter`, `JobUpdateParameter`
+  - Mark Tasks as preemptible for higher priority Tasks (requires Comms-Enabled or Single Tenant Pool).
+- Replaced comment (title, description, etc.) references of "low-priority" with "Spot/Low-Priority", to reflect new service behavior.
+  - No API change required.
+  - Low-Priority Compute Nodes (VMs) will continue to be used for User Subscription pools (and only User Subscription pools), as before.
+  - Spot Compute Nodes (VMs) will now be used for Batch Managed (and only Batch Managed pools) pools.
 
-### Bugs Fixed
+## 15.2.0 (Unreleased)
```
Comment (dpwatrous) [inline sdk/batch/Microsoft.Azure.Batch/CHANGELOG.md:17]: Let's just roll this change up to 15.3.0 if we never released 15.2.0 (and we can remove 15.2.0 from the release history)

File: sdk/batch/Microsoft.Azure.Batch/CHANGELOG.md [hunk 2]
```diff
@@ -1,16 +1,24 @@
 # Release History
 
-## 15.2.0-beta.1 (Unreleased)
+## 15.3.0 (2022-01-01)
```
Comment (dpwatrous) [inline sdk/batch/Microsoft.Azure.Batch/CHANGELOG.md:3]: We may want to bump this to the current date (this date is the rough release date, not the API version)

File: sdk/batch/Microsoft.Azure.Batch/Tools/ObjectModelCodeGenerator/Spec/HttpHeader.json [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+    {
+      "Key": {
+        "Type": "string",
+        "Name": "Name",
+        "SummaryComment": "The case-insensitive name of the header to be used while uploading output files.",
+        "RemarksComment": null,
+        "BoundAccess": "read,write",
+        "UnboundAccess": "read,write",
+        "ConstructorArgumentType": "Required",
+        "HideSetter": true
```
Comment (dpwatrous) [inline sdk/batch/Microsoft.Azure.Batch/Tools/ObjectModelCodeGenerator/Spec/HttpHeader.json:16]: Why are we hiding the setter here?

File: sdk/batch/Microsoft.Azure.Batch/Tools/ObjectModelCodeGenerator/Spec/HttpHeader.json [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+    {
+      "Key": {
+        "Type": "string",
+        "Name": "Value",
+        "SummaryComment": "The value of the header to be used while uploading output files.",
+        "RemarksComment": null,
+        "BoundAccess": "read,write",
+        "UnboundAccess": "read,write",
+        "ConstructorArgumentType": "Required",
+        "HideSetter": true
```
Comment (dpwatrous) [inline sdk/batch/Microsoft.Azure.Batch/Tools/ObjectModelCodeGenerator/Spec/HttpHeader.json:29]: (See my other question on hiding the setter)

**Review by dpwatrous — COMMENTED** [review]
Looks good overall - added just a few comments.

**Review by dpwatrous — APPROVED** [review]

**Pull request merged**